File: Source/dcc/Source/DemoApp/ConsumerCareDemoApp/ViewController.m
//
//  ViewController.m
//  ConsumerCareDemoApp
//
//  Created by Niharika Bundela on 3/22/17.
//  Copyright © 2017 Niharika Bundela. All rights reserved.
//

#import "ViewController.h"
#import "ConsumerCareMicroAppDependencies.h"
#import "ConsumerCareMicroAppInterface.h"
#import "ConsumerCareMicroAppLaunchInput.h"
#import <AppInfra/AppInfra.h>

@interface ViewController ()
@property (strong, nonatomic) ConsumerCareMicroAppInterface *dcInterface;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)LaunchApplication:(id)sender {
    ConsumerCareMicroAppDependencies *dcDependencies = [[ConsumerCareMicroAppDependencies alloc] init];
    dcDependencies.appInfra = [[AIAppInfra alloc]initWithBuilder:nil];;
    self.dcInterface = [[ConsumerCareMicroAppInterface alloc]initWithDependencies:dcDependencies andSettings:nil];
    ConsumerCareMicroAppLaunchInput *launchInput = [[ConsumerCareMicroAppLaunchInput alloc] init];
    UIViewController *childViewController = [self.dcInterface instantiateViewController:launchInput withErrorHandler:^(NSError *error) {
        NSLog(@"error happened");
    }];
    [self.navigationController pushViewController:childViewController animated:YES];
}

@end

File: Source/prg/Source/DemoApp/DemoApp/ViewController.swift
//
//  ViewController.swift
//  PPROneRoofDemo
//
//  Created by Abhishek on 21/03/17.
//  Copyright © 2017 Abhishek. All rights reserved.
//

import UIKit
import PhilipsProductRegistrationUApp
import UAPPFramework
import AppInfra
import PhilipsProductRegistration
import PhilipsPRXClient
import PhilipsRegistration

class ViewController: UIViewController {

    var prDemoDependencies: PPRInterfaceDependency?
    var prDemoInterface: PPRDemoFrameworkInterface?
    var prDemoLaunchInput: PPRDemoFrameworkLaunchInput?
    var launchOption: PPRUILaunchOption = PPRUILaunchOption.WelcomeScreen

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.

        let appdelegate: AppDelegate = UIApplication.shared.delegate as! AppDelegate

        self.prDemoDependencies = PPRInterfaceDependency()
        self.prDemoDependencies?.appInfra = appdelegate.appInfra
        self.prDemoDependencies?.userDataInterface = setUserDataInterface(appInfra: appdelegate.appInfra)
        self.prDemoInterface = PPRDemoFrameworkInterface(dependencies: self.prDemoDependencies!, andSettings: nil)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func setUserDataInterface(appInfra:AIAppInfra!) -> UserDataInterface!{
        let UserRegistrationDependencies = URDependencies()
        UserRegistrationDependencies.appInfra = appInfra
        let urInterface = URInterface(dependencies: UserRegistrationDependencies, andSettings: nil)
        return urInterface.userDataInterface()
    }

    @IBAction func launchProductRegistrationDemo(_ sender: AnyObject) {
        let launchInput = PPRDemoFrameworkLaunchInput()

        let viewController: UIViewController? = self.prDemoInterface?.instantiateViewController(launchInput, withErrorHandler: {(_ error: Error?) -> Void in
            print("Update UI")
        })

        self.navigationController?.pushViewController(viewController!, animated: true)
    }
}

